lib/manager/components/FeedSourceSettings.js

```diff
@@ -57,67 +59,98 @@ export default class FeedSourceSettings extends Component<Props> {
-                  </LinkContainer>
-                )
-              })}
-            </ListGroup>
-          </Panel>
-        </Col>
-        <Col xs={6} />
-        {!resourceType
-          ? <GeneralSettings
+
+    const tabs = [
+      {
+        condition: true,
+        render: () => (
+          <GeneralSettings
             confirmDeleteFeedSource={this.props.confirmDeleteFeedSource}
             disabled={disabled}
             feedSource={feedSource}
             project={project}
             updateFeedSource={this.props.updateFeedSource}
             user={user}
           />
-          : showTransformationsTab
-            ? <FeedTransformationSettings
-              disabled={disabled}
-              feedSource={feedSource}
-              project={project}
-              updateFeedSource={this.props.updateFeedSource}
-              user={user}
-            />
-            : <Col xs={7}>
+        ),
+        subPath: '',
+        title: 'General'
+      },
+      {
+        condition: true,
+        render: () => (
+          <FeedTransformationSettings
+            disabled={disabled}
+            feedSource={feedSource}
+            project={project}
+            updateFeedSource={this.props.updateFeedSource}
+            user={user}
+          />
+        ),
+        subPath: 'transformations',
+        title: 'Feed Transformations'
+      },
+      {
+        condition: isExtensionEnabled('mtc'),
+        render: () => (
+          <AutoPublishSettings
+            disabled={disabled}
+            feedSource={feedSource}
+            updateFeedSource={this.props.updateFeedSource}
+          />
+        ),
+        subPath: 'autopublish',
+        title: 'Auto-publish'
+      },
+      ...Object.keys(feedSource.externalProperties || {}).map(resourceType => {
+        const resourceLowerCase = resourceType.toLowerCase()
+        return {
+          condition: true,
+          render: () => (
+            <Col xs={7}>
               <ExternalPropertiesTable
-                resourceType={resourceType}
-                feedSource={feedSource}
                 editingIsDisabled={disabled}
+                feedSource={feedSource}
                 isProjectAdmin={isProjectAdmin}
                 resourceProps={feedSource.externalProperties
                   ? feedSource.externalProperties[resourceType]
                   : {}
                 }
+                resourceType={resourceType.toUpperCase()}
                 updateExternalFeedResource={updateExternalFeedResource} />
             </Col>
+          ),
+          subPath: resourceLowerCase,
+          title: `${toSentenceCase(resourceType)} properties`
         }
+      })
+    ]
+
+    const activeTab = tabs.find(tab => resourceType === tab.subPath) || tabs[0]
+
+    return (
+      <Row>
+        <Col xs={3}>
+          {/* Side panel */}
+          <Panel>
+            <ListGroup fill>
```

Comment:
I'm loving this refactor, but there is a regression! The general tab is no longer highlighted in blue when first loading the page.

Reply:
Yes, thanks, fixed in https://github.com/ibi-group/datatools-ui/pull/746/commits/8c18b7adf5ab59e5a6ca04298e6ff2aae40c244a!